src/core/etl/tests/Flow/ETL/Tests/Unit/Row/Entry/StructureEntryTest.php

```diff
@@ -18,65 +22,140 @@ public static function is_equal_data_provider() : \Generator
+            ),
+            new StructureEntry(
+                'name',
+                ['1' => 1, '2' => 2, '3' => 3],
+                new StructureType(new StructureElement('1', ScalarType::integer()), new StructureElement('2', ScalarType::string()), new StructureElement('3', ScalarType::string()))
+            ),
         ];
         yield 'equal names and equal simple same array entries' => [
             true,
-            new StructureEntry('name', new ArrayEntry('json', ['foo' => ['bar' => 'baz']])),
-            new StructureEntry('name', new ArrayEntry('json', ['foo' => ['bar' => 'baz']])),
+            new StructureEntry(
+                'name',
+                ['json' => ['foo' => ['bar' => 'baz']]],
+                new StructureType(new StructureElement('json', new MapType(MapKey::string(), MapValue::map(new MapType(MapKey::string(), MapValue::string())))))
+            ),
+            new StructureEntry(
+                'name',
+                ['json' => ['foo' => ['bar' => 'baz']]],
+                new StructureType(new StructureElement('json', new MapType(MapKey::string(), MapValue::map(new MapType(MapKey::string(), MapValue::string())))))
+            ),
         ];
         yield 'equal names and equal simple same collection entries' => [
             true,
-            new StructureEntry('name', new StructureEntry('json', new IntegerEntry('1', 1), new IntegerEntry('2', 2), new IntegerEntry('3', 3))),
-            new StructureEntry('name', new StructureEntry('json', new IntegerEntry('1', 1), new IntegerEntry('2', 2), new IntegerEntry('3', 3))),
+            new StructureEntry(
+                'name',
+                ['json' => ['1' => 1, '2' => 2, '3' => 3]],
+                new StructureType(new StructureElement('json', new ArrayType()))
+            ),
+            new StructureEntry(
+                'name',
+                ['json' => ['1' => 1, '2' => 2, '3' => 3]],
+                new StructureType(new StructureElement('json', new ArrayType()))
+            ),
         ];
         yield 'equal names and equal simple different collection entries' => [
             false,
-            new StructureEntry('name', new StructureEntry('json', new IntegerEntry('5', 5), new IntegerEntry('2', 2), new IntegerEntry('3', 3))),
-            new StructureEntry('name', new StructureEntry('json', new IntegerEntry('1', 1), new IntegerEntry('2', 2), new IntegerEntry('3', 3))),
+            new StructureEntry(
+                'name',
+                ['json' => ['5' => 5, '2' => 2, '1' => 1]],
+                new StructureType(new StructureElement('json', new ArrayType()))
+            ),
+            new StructureEntry(
+                'name',
+                ['json' => ['1' => 1, '2' => 2, '3' => 3]],
+                new StructureType(new StructureElement('json', new ArrayType()))
+            ),
```

Comment:
We going to need a few more tests here since we are no longer using Entries. 

- using nullable struct elements
- missing struct elements
- too many struct elements 
- invalid types struct elements